### PR TITLE
PRM-2600 Rework of unit test data

### PR DESCRIPTION
**Pull request by AndrewRButcher**

### CODE REVIEW

File: test/views/components/LateSubmissionPenaltySummaryCardSpec.scala [hunk 1]
```diff
@@ -707,14 +707,13 @@ class LateSubmissionPenaltySummaryCardSpec extends SpecBase with ViewBehaviours
       }
 
       "have the appeal status for REJECTED" in {
-        docWithAppealedPointRejected.select("dt").text().contains("Point due to expire") shouldBe true
```
Comment (LewisMitchell) [inline test/views/components/LateSubmissionPenaltySummaryCardSpec.scala:710]: Should we have a check for this? I guess the penalty is a charge, right?

File: test/views/components/LatePaymentPenaltySummaryCardSpec.scala [hunk 2]
```diff
@@ -311,7 +311,7 @@ class LatePaymentPenaltySummaryCardSpec extends SpecBase with ViewBehaviours {
 
       "display the penalty amount and the date the VAT was due (with padded zero for whole tenths)" in {
         docWithAdditionalPenaltyTenthsOfPence.select("h4").get(0).ownText shouldBe "£123.40 penalty"
-        docWithAdditionalPenaltyTenthsOfPence.select("h4 span").text shouldBe "for late payment of charge due on 1 January 2021"
+        docWithAdditionalPenaltyTenthsOfPence.select("h4 span").text shouldBe "for late payment of charge due on 8 June 2021"
```
Comment (LewisMitchell) [inline test/views/components/LatePaymentPenaltySummaryCardSpec.scala:314]: Shouldn't the due date be 7 June 2021?

File: test/base/testData/LPPDetailsTestData.scala [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+import models.appealInfo.{AppealInformationType, AppealLevelEnum, AppealStatusEnum}
+import models.lpp._
+
+import java.time.LocalDate
+
+trait LPPDetailsTestData {
+  val sampleDateLPP: LocalDate = LocalDate.of(2021, 1, 1)
+  val penaltyChargeCreationDate: LocalDate = LocalDate.of(2021, 1, 6)
+  val comsDate: LocalDate = penaltyChargeCreationDate
```
Comment (LewisMitchell) [inline test/base/testData/LPPDetailsTestData.scala:27]:
```suggestion
  val commsDate: LocalDate = penaltyChargeCreationDate
```

File: test/base/testData/LSPDetailsTestData.scala [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        taxPeriodDueDate = Some(taxPeriodDue),
+        returnReceiptDate = Some(receiptDate),
+        taxReturnStatus = TaxReturnStatusEnum.Fulfilled
+      )
+    ))
+  )
+
+  val sampleRemovedPenaltyPoint: LSPDetails = sampleLateSubmissionPoint.copy(
+    penaltyStatus = LSPPenaltyStatusEnum.Inactive,
+    expiryReason = Some(ExpiryReasonEnum.Adjustment)
```
Comment (LewisMitchell) [inline test/base/testData/LSPDetailsTestData.scala:101]: Would this have a FAPIndicator set?

File: test/viewmodels/SummaryCardHelperSpec.scala [hunk 5]
```diff
@@ -77,9 +77,9 @@ class SummaryCardHelperSpec extends SpecBase with ImplicitDateFormatter {
       amountDue = 1001.45,
       isPenaltyPaid = true,
       penaltyCategory = LPP1,
-      dueDate = "1\u00A0January\u00A02021",
-      taxPeriodStartDate = sampleDate.toString,
-      taxPeriodEndDate = sampleDate.toString,
+      dueDate = "8\u00A0June\u00A02021",
```
Comment (LewisMitchell) [inline test/viewmodels/SummaryCardHelperSpec.scala:80]:
```suggestion
      dueDate = "7\u00A0June\u00A02021",
```

File: test/viewmodels/SummaryCardHelperSpec.scala [hunk 6]
```diff
@@ -107,9 +107,9 @@ class SummaryCardHelperSpec extends SpecBase with ImplicitDateFormatter {
       isVatPaid = true,
       amountDue = 1001.45,
       penaltyCategory = LPP2,
-      dueDate = "1\u00A0January\u00A02021",
-      taxPeriodStartDate = sampleDate.toString,
-      taxPeriodEndDate = sampleDate.toString,
+      dueDate = "8\u00A0June\u00A02021",
```
Comment (LewisMitchell) [inline test/viewmodels/SummaryCardHelperSpec.scala:110]:
```suggestion
      dueDate = "7\u00A0June\u00A02021",
```

File: test/viewmodels/SummaryCardHelperSpec.scala [hunk 7]
```diff
@@ -329,9 +326,9 @@ class SummaryCardHelperSpec extends SpecBase with ImplicitDateFormatter {
           Seq(
             helper.summaryListRow(
               period,
-              Html(vatPeriodValue(dateToString(sampleDate), dateToString(sampleDate)))
+              Html(vatPeriodValue(dateToString(sampleDateLSP), dateToString(sampleDateLSP)))
```
Comment (LewisMitchell) [inline test/viewmodels/SummaryCardHelperSpec.scala:329]: Is it possible to use taxPeriodStart and taxPeriodEnd here?

File: test/viewmodels/SummaryCardHelperSpec.scala [hunk 8]
```diff
@@ -329,9 +326,9 @@ class SummaryCardHelperSpec extends SpecBase with ImplicitDateFormatter {
...
             ),
-            helper.summaryListRow(returnDue, Html(dateToString(sampleDate))),
+            helper.summaryListRow(returnDue, Html(dateToString(sampleDateLSP))),
```
Comment (LewisMitchell) [inline test/viewmodels/SummaryCardHelperSpec.scala:331]: Is it possible to use taxPeriodDue here?

File: test/viewmodels/SummaryCardHelperSpec.scala [hunk 9]
```diff
@@ -340,16 +337,16 @@ class SummaryCardHelperSpec extends SpecBase with ImplicitDateFormatter {
           isReturnSubmitted = false,
           isThresholdPoint = true,
           totalPenaltyAmount = 200,
-          dueDate = Some(dateToString(sampleDate))
+          dueDate = Some(dateToString(sampleDateLSP))
```
Comment (LewisMitchell) [inline test/viewmodels/SummaryCardHelperSpec.scala:340]: Is it possible to use taxPeriodDue here?

File: test/viewmodels/SummaryCardHelperSpec.scala [hunk 10]
```diff
@@ -340,16 +337,16 @@ class SummaryCardHelperSpec extends SpecBase with ImplicitDateFormatter {
...
+        val pointToPassIn: LSPDetails = sampleLateSubmissionPenaltyCharge.copy(penaltyOrder = "1", penaltyCategory = LSPPenaltyCategoryEnum.Threshold,
           lateSubmissions = Some(
             Seq(
               LateSubmission(
-                taxPeriodStartDate = Some(sampleDate),
-                taxPeriodEndDate = Some(sampleDate),
-                taxPeriodDueDate = Some(sampleDate),
+                taxPeriodStartDate = Some(sampleDateLSP),
+                taxPeriodEndDate = Some(sampleDateLSP),
+                taxPeriodDueDate = Some(sampleDateLSP),
```
Comment (LewisMitchell) [inline test/viewmodels/SummaryCardHelperSpec.scala:349]: Is it possible to use taxPeriodStart, taxPeriodEnd and taxPeriodDue here?

File: test/viewmodels/SummaryCardHelperSpec.scala [hunk 11]
```diff
@@ -364,70 +361,70 @@ class SummaryCardHelperSpec extends SpecBase with ImplicitDateFormatter {
...
             Seq(
               helper.summaryListRow(
                 period,
-                Html(vatPeriodValue(dateToString(sampleDate), dateToString(sampleDate)))
+                Html(vatPeriodValue(dateToString(sampleDateLSP), dateToString(sampleDateLSP)))
               ),
-              helper.summaryListRow(returnDue, Html(dateToString(sampleDate))),
-              helper.summaryListRow(returnSubmitted, Html(dateToString(sampleDate))),
+              helper.summaryListRow(returnDue, Html(dateToString(sampleDateLSP))),
+              helper.summaryListRow(returnSubmitted, Html(dateToString(sampleDateLSP))),
```
Comment (LewisMitchell) [inline test/viewmodels/SummaryCardHelperSpec.scala:382]: Is it possible to use taxPeriodStart, taxPeriodEnd, taxPeriodDue and receiptDate here?

File: test/viewmodels/SummaryCardHelperSpec.scala [hunk 12]
```diff
@@ -364,70 +361,70 @@ class SummaryCardHelperSpec extends SpecBase with ImplicitDateFormatter {
...
               helper.summaryListRow(pointExpiration, Html(dateTimeToMonthYearString(LocalDateTime.now)))
             ),
             Tag(content = Text("active")),
             "3",
             "12345678901234",
             isReturnSubmitted = true,
-            dueDate = Some(dateToString(sampleDate))
+            dueDate = Some(dateToString(sampleDateLSP))
```
Comment (LewisMitchell) [inline test/viewmodels/SummaryCardHelperSpec.scala:389]: Is it possible to use taxPeriodDue?

File: test/viewmodels/SummaryCardHelperSpec.scala [hunk 13]
```diff
@@ -364,70 +361,70 @@ class SummaryCardHelperSpec extends SpecBase with ImplicitDateFormatter {
...
               Seq(
                 helper.summaryListRow(
                   period,
-                  Html(vatPeriodValue(dateToString(sampleDate), dateToString(sampleDate)))
+                  Html(vatPeriodValue(dateToString(sampleDateLSP), dateToString(sampleDateLSP)))
                 ),
-                helper.summaryListRow(returnDue, Html(dateToString(sampleDate))),
-                helper.summaryListRow(returnSubmitted, Html(dateToString(sampleDate))),
+                helper.summaryListRow(returnDue, Html(dateToString(sampleDateLSP))),
+                helper.summaryListRow(returnSubmitted, Html(dateToString(sampleDateLSP))),
```
Comment (LewisMitchell) [inline test/viewmodels/SummaryCardHelperSpec.scala:398]: Is it possible to use taxPeriodStart, taxPeriodEnd, taxPeriodDue and receiptDate here?

File: test/viewmodels/SummaryCardHelperSpec.scala [hunk 14]
```diff
@@ -364,70 +361,70 @@ class SummaryCardHelperSpec extends SpecBase with ImplicitDateFormatter {
...
                 helper.summaryListRow(pointExpiration, Html(dateTimeToMonthYearString(LocalDateTime.now)))
               ),
               Tag(content = Text("active")),
               "2",
               "12345678901234",
               isReturnSubmitted = true,
-              dueDate = Some(dateToString(sampleDate))
+              dueDate = Some(dateToString(sampleDateLSP))
```
Comment (LewisMitchell) [inline test/viewmodels/SummaryCardHelperSpec.scala:405]: Is it possible to use taxPeriodDue?

File: test/viewmodels/SummaryCardHelperSpec.scala [hunk 15]
```diff
@@ -364,70 +361,70 @@ class SummaryCardHelperSpec extends SpecBase with ImplicitDateFormatter {
...
               Seq(
                 helper.summaryListRow(
                   period,
-                  Html(vatPeriodValue(dateToString(sampleDate), dateToString(sampleDate)))
+                  Html(vatPeriodValue(dateToString(sampleDateLSP), dateToString(sampleDateLSP)))
                 ),
-                helper.summaryListRow(returnDue, Html(dateToString(sampleDate))),
-                helper.summaryListRow(returnSubmitted, Html(dateToString(sampleDate))),
+                helper.summaryListRow(returnDue, Html(dateToString(sampleDateLSP))),
+                helper.summaryListRow(returnSubmitted, Html(dateToString(sampleDateLSP))),
```
Comment (LewisMitchell) [inline test/viewmodels/SummaryCardHelperSpec.scala:414]: Is it possible to use taxPeriodStart, taxPeriodEnd, taxPeriodDue and receiptDate here?

File: test/viewmodels/SummaryCardHelperSpec.scala [hunk 16]
```diff
@@ -364,70 +361,70 @@ class SummaryCardHelperSpec extends SpecBase with ImplicitDateFormatter {
...
                 helper.summaryListRow(pointExpiration, Html(dateTimeToMonthYearString(LocalDateTime.now)))
               ),
               Tag(content = Text("active")),
               "1",
               "12345678901234",
               isReturnSubmitted = true,
-              dueDate = Some(dateToString(sampleDate))
+              dueDate = Some(dateToString(sampleDateLSP))
```
Comment (LewisMitchell) [inline test/viewmodels/SummaryCardHelperSpec.scala:421]: Is it possible to use taxPeriodDue?

File: test/viewmodels/SummaryCardHelperSpec.scala [hunk 17]
```diff
@@ -364,70 +361,70 @@ class SummaryCardHelperSpec extends SpecBase with ImplicitDateFormatter {
...
             ),
             LateSubmissionPenaltySummaryCard(
               Seq(
                 helper.summaryListRow(
                   period,
-                  Html(vatPeriodValue(dateToString(sampleDate), dateToString(sampleDate)))
+                  Html(vatPeriodValue(dateToString(sampleDateLSP), dateToString(sampleDateLSP)))
```
Comment (LewisMitchell) [inline test/viewmodels/SummaryCardHelperSpec.scala:427]: Is it possible to use taxPeriodStart and taxPeriodEnd?

File: test/viewmodels/SummaryCardHelperSpec.scala [hunk 18]
```diff
@@ -436,7 +433,7 @@ class SummaryCardHelperSpec extends SpecBase with ImplicitDateFormatter {
               "12345678901234",
               isReturnSubmitted = true,
               isAddedOrRemovedPoint = true,
-              dueDate = Some(dateToString(sampleDate))
+              dueDate = Some(dateToString(sampleDateLSP))
```
Comment (LewisMitchell) [inline test/viewmodels/SummaryCardHelperSpec.scala:436]: Is it possible to use taxPeriodDue?

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

File: test/views/components/LateSubmissionPenaltySummaryCardSpec.scala (hunk 1 above)
Comment (LewisMitchell) [reply test/views/components/LateSubmissionPenaltySummaryCardSpec.scala:710]: What do you think to this?

Comment (AndrewRButcher) [reply test/views/components/LateSubmissionPenaltySummaryCardSpec.scala:710]: I am not sure what you are referring too/ what the issue is?

Comment (LewisMitchell) [reply test/views/components/LateSubmissionPenaltySummaryCardSpec.scala:710]: This line was removed as part of your PR, I think it is a useful check we should have

Comment (AndrewRButcher) [reply test/views/components/LateSubmissionPenaltySummaryCardSpec.scala:710]: As it is a financial penalty it should not have an expiry date, I have updated the case data to be more clear on the penalty type

**Pull request merged**